code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -58,20 +102,34 @@
 		M.adjust_fire_stacks(reac_volume / 15)
 	return ..()
 
+/datum/reagent/consumable/ethanol/on_mob_delete(mob/living/M)
+	if(iscarbon(M))
+		var/mob/living/carbon/human/H = M
+		spawn(300) //30 seconds
+			H.drunk_state = NOT_DRUNK //Shouldn't be a problem if they have more alcohol in their system since it'll immediately be set back
```

Comment:
If someone drinks two drinks, with different quantities, wouldn't this cause drunkeness to reset after the first one metabolizes?